esmvalcore/cmor/check.py

```diff
@@ -475,21 +475,29 @@ def _check_time_coord(self):
             coord.units = cf_units.Unit(coord.units.origin, simplified_cal)
 
             attrs = self._cube.attributes
-            branch_child = 'branch_time_in_child'
-            if branch_child in attrs:
-                attrs[branch_child] = old_units.convert(attrs[branch_child],
-                                                        coord.units)
-
-            parent_time = 'parent_time_units'
-            if parent_time in attrs:
-                parent_units = cf_units.Unit(attrs[parent_time],
-                                             simplified_cal)
-                attrs[parent_time] = 'days since 1850-1-1 00:00:00'
-
-                branch_parent = 'branch_time_in_parent'
-                if branch_parent in attrs:
-                    attrs[branch_parent] = parent_units.convert(
-                        attrs[branch_parent], coord.units)
+            try:
+                parent_time = 'parent_time_units'
+                if parent_time in attrs:
+                    parent_units = cf_units.Unit(attrs[parent_time],
+                                                 simplified_cal)
+                    attrs[parent_time] = 'days since 1850-1-1 00:00:00'
+
+                    branch_parent = 'branch_time_in_parent'
+                    if branch_parent in attrs:
+                        attrs[branch_parent] = parent_units.convert(
+                            attrs[branch_parent], coord.units)
+
+                branch_child = 'branch_time_in_child'
+                if branch_child in attrs:
+                    attrs[branch_child] = old_units.convert(
+                        attrs[branch_child], coord.units)
+            except ValueError:
```

Comment:
For `try`/`except` clauses, limit the `try` clause to the absolute minimum amount of code necessary, this avoids masking bugs. See [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)